src/components/dialogs/delete-voltage-level-on-line.js

```diff
@@ -0,0 +1,198 @@
+import { useParams } from 'react-router-dom';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { useInputForm, useTextValue } from './inputs/input-hooks';
+import {
+    gridItem,
+    GridSection,
+    compareById,
+    sanitizeString,
+} from './dialogUtils';
+import { deleteVoltageLevelOnLine } from '../../utils/rest-api';
+import PropTypes from 'prop-types';
+import { useAutocompleteField } from './inputs/use-autocomplete-field';
+import ModificationDialog from './modificationDialog';
+
+const getId = (e) => e?.id || (typeof e === 'string' ? e : '');
+
+/**
+ * Dialog to delete a voltage on a line.
+ * @param {Boolean} open Is the dialog open ?
+ * @param {EventListener} onClose Event to close the dialog
+ * @param lineOptionsPromise Promise handling list of network lines
+ * @param currentNodeUuid the currently selected tree node
+ * @param editData record to edit
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+const DeleteVoltageLevelOnLineDialog = ({
+    lineOptionsPromise,
+    currentNodeUuid,
+    editData,
+    ...dialogProps
+}) => {
+    const studyUuid = decodeURIComponent(useParams().studyUuid);
+
+    const { snackError } = useSnackMessage();
+
+    const inputForm = useInputForm();
+
+    const [formValues, setFormValues] = useState(undefined);
+
+    const [lineOptions, setLineOptions] = useState([]);
+
+    const [loadingLineOptions, setLoadingLineOptions] = useState(true);
+
+    useEffect(() => {
+        if (editData) {
+            setFormValues(editData);
+        }
+    }, [editData]);
+
+    useEffect(() => {
+        if (!lineOptionsPromise) return;
+        lineOptionsPromise.then((values) => {
+            setLineOptions(values);
+            setLoadingLineOptions(false);
+        });
+    }, [lineOptionsPromise]);
+
+    const formValueLineTo1AttachId = useMemo(() => {
+        return formValues?.lineToAttachTo1Id
+            ? { id: formValues?.lineToAttachTo1Id }
+            : { id: '' };
+    }, [formValues]);
+
+    const formValueLineTo2AttachId = useMemo(() => {
+        return formValues?.lineToAttachTo2Id
+            ? { id: formValues?.lineToAttachTo2Id }
+            : { id: '' };
+    }, [formValues]);
+
+    const [lineToAttachTo1, lineToAttachTo1Field] = useAutocompleteField({
+        id: 'lineToAttachTo1',
+        label: 'ID',
+        validation: { isFieldRequired: true },
+        inputForm: inputForm,
+        values: lineOptions?.sort(compareById),
+        allowNewValue: true,
+        getLabel: getId,
+        defaultValue:
+            lineOptions.find(
+                (value) => value.id === formValues?.lineToAttachTo1Id
+            ) || formValueLineTo1AttachId,
+        loading: loadingLineOptions,
+    });
+
+    const [lineToAttachTo2, lineToAttachTo2Field] = useAutocompleteField({
+        id: 'lineToAttachTo2',
+        label: 'ID',
+        validation: { isFieldRequired: true },
```

Comment:
surprising, isFieldRequired does not work when I try to validate the form without changing this filed.

Reply:
this is a separate and known issue - dont address it